File: php/features.php
<?php
declare(strict_types=1);

// Cataas SDK feature factory

require_once __DIR__ . '/feature/BaseFeature.php';
require_once __DIR__ . '/feature/TestFeature.php';


class CataasFeatures
{
    public static function make_feature(string $name)
    {
        switch ($name) {
            case "base":
                return new CataasBaseFeature();
            case "test":
                return new CataasTestFeature();
            default:
                return new CataasBaseFeature();
        }
    }
}

File: py/features.py
# Cataas SDK feature factory

from feature.base_feature import CataasBaseFeature
from feature.test_feature import CataasTestFeature


def _make_feature(name):
    features = {
        "base": lambda: CataasBaseFeature(),
        "test": lambda: CataasTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
